### PR TITLE
Fix the NotificationToast transition issues

### PR DESCRIPTION
**Pull request by amelako**
## Purpose

Regarding the transitions when the toast appears, its a bit hard and jumps a bit in its height. There was an issue with disappear transitions too, where the toast is animated to half the height and then fades out.

## Approach and changes

To fix the issues following changes were done:
- Initialize the toast visibility to false so that the toast animates in.
- Added an additional `div` around the toast content to fix the height transition issue. The padding needs to be on the inner `div`.
- Renamed the timeout to transition duration for better readability.

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: .changeset/cool-apes-live.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sumup/circuit-ui': patch
+---
+
+Fixed the NotificationToast transitions issue, so the toast appear with smoother fade + height transition.
```
Comment (connor-baer) [inline .changeset/cool-apes-live.md:5]:
```suggestion
Fixed the issue of the NotificationToast transition, now the toast appears with a smoother fade and height transition.
```

File: packages/circuit-ui/components/NotificationToast/NotificationToast.tsx [hunk 2]
```diff
@@ -79,30 +79,36 @@ const iconMap = {
...
-  align-items: center;
-  position: relative;
   background-color: ${theme.colors.bodyBg};
-  padding: ${theme.spacings.kilo} ${theme.spacings.mega};
   border-radius: ${theme.borderRadius.byte};
   border: ${theme.borderWidth.mega} solid ${theme.colors[colorMap[variant]]};
   overflow: hidden;
-  transition: opacity 200ms ease-in-out, height 200ms ease-in-out,
-    visibility 200ms ease-in-out;
+  will-change: height;
```
Comment (robinmetral) [inline packages/circuit-ui/components/NotificationToast/NotificationToast.tsx:94]: Did you notice a change when adding this? Curious

**Review by robinmetral — APPROVED** [review]
Noice 🐛

File: packages/circuit-ui/components/NotificationToast/NotificationToast.tsx (hunk 2 above)
Comment (connor-baer) [reply packages/circuit-ui/components/NotificationToast/NotificationToast.tsx:94]: I think this would only be noticeable on older hardware.

**Review by connor-baer — APPROVED** [review]

**Pull request merged**